astropy/units/core.py

```diff
@@ -2465,7 +2472,12 @@ def is_unity(self) -> bool:
 ]
 
 
-def _add_prefixes(u, excludes=[], namespace=None, prefixes=False):
+def _add_prefixes(
+    u: NamedUnit,
+    excludes: Collection[str] = (),
+    namespace: MutableMapping[str, object] | None = None,
+    prefixes: bool | Iterable[tuple[Iterable[str], Iterable[str], Complex]] = False,
```

Comment:
Instead of `Complex`, I'd go for `UnitScale`, as that immediately gives the purpose.

Reply:
Perhaps we should rename `Complex` to `SupportsUnitScale` (and `Real` to `SupportsUnitPower`). That could also help distinguish them from `numbers.Real` and `numbers.Complex`.

Reply:
I'm in favour of changing the name; `Complex` as a placeholder for "any numeric scalar" is not very helpful for human readers. `Supports*` feels a bit off, I guess one wants to indicate it is convertible to a power or scale. Or `RealScalar` and `NumericScalar`? It is the `complex` part that always throws me off - that so strongly suggests that the number must be complex instead of just allowing it to be complex.

Reply:
`Supports*` seems to be the Python standard library convention for indicating the implementation of some specific dunder method (e.g. `SupportsIndex` implements  `__index__`). `UnitScale` and `UnitPower` don't have corresponding dunder methods, so something like `SupportsUnitScale` would indeed be misleading.

Our glossary has an entry for `-like`: https://github.com/astropy/astropy/blob/80aa67d2cc7ddb41a63d00d231b2792aafca77fa/docs/glossary.rst?plain=1#L15-L19
That suggests that `Complex` should be renamed to `UnitScaleLike`.

Reply:
Yes, the `*Like` one works for me!

p.s. `SupportsIndexing` does correctly indicate that it supports some specific action; as you note, it doesn't really apply as well here.

Reply:
Maybe we should consider using similar naming conventions to those in scipy-stubs. Which we might be, already?

Reply:
I'm not familiar with the naming conventions `scipy-stubs` uses, but I do know that our documentation already uses `*-like`, which in type names naturally maps to `*Like`.